src/attached_probe.cpp

```diff
@@ -223,16 +225,142 @@ std::string AttachedProbe::sanitise(const std::string &str)
   return std::regex_replace(str, std::regex("[^A-Za-z0-9_]"), "_");
 }
 
-uint64_t AttachedProbe::offset() const
+struct symbol {
+  std::string name;
+  uint64_t    start;
+  uint64_t    size;
+  uint64_t    address;
+};
+
+static int sym_name_cb(const char *symname, uint64_t start,
+                       uint64_t size, void *p)
 {
-  bcc_symbol sym;
-  int err = bcc_resolve_symname(probe_.path.c_str(), probe_.attach_point.c_str(),
-      probe_.loc, 0, nullptr, &sym);
+  struct symbol *sym = static_cast<struct symbol*>(p);
 
-  if (err)
-    throw std::runtime_error("Could not resolve symbol: " + probe_.path + ":" + probe_.attach_point);
+  if (sym->name == symname)
+  {
+    sym->start = start;
+    sym->size  = size;
+    return -1;
+  }
+
+  return 0;
+}
+
+static int sym_address_cb(const char *symname, uint64_t start,
+                          uint64_t size, void *p)
+{
+  struct symbol *sym = static_cast<struct symbol*>(p);
+
+  if (sym->address >= start && sym->address < (start + size))
+  {
+    sym->start = start;
+    sym->size  = size;
+    sym->name  = symname;
+    return -1;
+  }
 
-  return sym.offset;
+  return 0;
+}
```

Comment:
Should we use the bcc helpers instead? https://github.com/iovisor/bcc/blob/master/src/cc/bcc_syms.h#L63-L69

Reply:
ok, will check on those helpers

Reply:
..so they all work over symcache, where the only we could use is BuildSyms class.
But they return symbol in 'struct bcc_symbol' which provides file offset, while we need
the size and vaddr.

Perhaps I could submit a change to bcc to get all missing info, and we could devise
some common way of getting symbol info through bpftrace code.. I think I've already
saw some issue with resolving symbols... thoughts @danobi ?

Reply:
Hmm extending bcc takes quite a while for everyone to propagate through distro repos. Probably for now we could use our own callbacks and also send bcc a patch (up to you).

> I think I've already saw some issue with resolving symbols

On bcc side or bpftrace side?

Reply:
I recall seeing some issue about unresolved symbols in bpftrace output,
can't find it now with a quick look..